### PR TITLE
add files for html-css popup

### PR DESCRIPTION
**Pull request by DmytryjK**
### HTML-CSS-Popup

[Demo](https://dmytryjk.github.io/Hooli-style/) |
[Code base](https://github.com/DmytryjK/Hooli-style-popup)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/DmytryjK/HTML-CSS-Popup/css/components/fonts.css [hunk 1]
```diff
@@ -0,0 +1,14 @@
+@font-face {
+  font-family: "inter-regular";
+  src: url("../fonts/Inter2/converted/Inter-Regular.eot");
```
Comment (A-Ostrovnyy) [inline submissions/DmytryjK/HTML-CSS-Popup/css/components/fonts.css:3]: For modern browsers enough `.woff` and `woff2` fonts.

File: submissions/DmytryjK/HTML-CSS-Popup/css/components/fonts.css [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  font-weight: 400;
+  font-style: normal;
+}
+@font-face {
+  font-family: "inter-bold";
+  src: url("../fonts/Inter2/converted/Inter-Bold.eot");
+  src: url("../fonts/Inter2/converted/Inter-Bold.woff") format("woff"), url("../fonts/Inter2/converted/Inter-Bold.otf") format("opentype"), url("../fonts/Inter2/converted/Inter-Bold.ttff") format("truetype");
+  font-weight: 700;
+  font-style: bold;
+}
```
Comment (A-Ostrovnyy) [inline submissions/DmytryjK/HTML-CSS-Popup/css/components/fonts.css:14]: Add an empty line at the end of the file. [Reason](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline?noredirect=1&lq=1)

File: submissions/DmytryjK/HTML-CSS-Popup/css/style.css [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+  justify-content: space-between;
+  align-items: center;
+  max-width: 1140px;
+  margin: 0 auto;
+  padding: 0 20px;
+}
+
+.header {
+  background-color: #252525;
+  /*label & inputs for submenu*/
```
Comment (A-Ostrovnyy) [inline submissions/DmytryjK/HTML-CSS-Popup/css/style.css:32]: Avoid committing commented lines.

File: submissions/DmytryjK/HTML-CSS-Popup/css/components/header.css [hunk 4]
```diff
@@ -0,0 +1,7 @@
+.header__logo-img {
```
Comment (A-Ostrovnyy) [inline submissions/DmytryjK/HTML-CSS-Popup/css/components/header.css:1]: Why do you have styles for header in the next file if this file with header styles?

File: submissions/DmytryjK/HTML-CSS-Popup/index.html [hunk 5]
```diff
@@ -0,0 +1 @@
...ader__popup-link" href="#"><picture><source srcset="images/WEBP/promo.webp" type="image/webp"><source srcset="images/promo.png" type="image/jpeg"><img class="header__popup-img header__popup-img_width" src="images/promo.png" alt="promo icon"></picture><span>Promo</span></a></li><li class="header__popup-item header__more-items"><a class="header__popup-link" href="#"><picture><source srcset="images/mic.png" type="image/jpeg"><img class="header__popup-img header__popup-img_width" src="images/mic.png" alt="mic icon"></picture><span>Mic</span></a></li><li class="header__popup-item header__more-items"><a class="header__popup-link" href="#"><picture><source srcset="images/collection.png" type="image/jpeg"><img class="header__popup-img header__popup-img_width" src="images/search.png" alt="collection icon"></picture><span>Search</span></a></li></ul></label></li></ul></label></li><li class="header__items"><a class="header__links" href="#"><img class="header__bell-icon" src="images/bell.png" alt="bell img"></a></li><li class="header__items"><a class="header__links" href="#"><img class="header__user-avatar" src="images/man.svg" alt="user avatar"></a></li></ul></nav></div></header></body></html>
```
Comment (A-Ostrovnyy) [inline submissions/DmytryjK/HTML-CSS-Popup/index.html:1]: Fix indent of file?

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Nice work!

File: submissions/DmytryjK/HTML-CSS-Popup/css/components/header.css (hunk 4 above)
Comment (DmytryjK) [reply submissions/DmytryjK/HTML-CSS-Popup/css/components/header.css:1]: Initially, I broke the styles into modules, a separate _header_, _fonts_, _variables_. I used to gulp for compile all modules to 1 main style.css file. Apparently somesing went wrong, when I uploaded them to the server.
Should I re-upload the sources, or just leave 1 main file style.css, and delete the rest?

File: submissions/DmytryjK/HTML-CSS-Popup/index.html (hunk 5 above)
Comment (DmytryjK) [reply submissions/DmytryjK/HTML-CSS-Popup/index.html]: fixed

File: submissions/DmytryjK/HTML-CSS-Popup/css/components/fonts.css (hunk 1 above)
Comment (DmytryjK) [reply submissions/DmytryjK/HTML-CSS-Popup/css/components/fonts.css]: fixed

File: submissions/DmytryjK/HTML-CSS-Popup/css/components/fonts.css (hunk 2 above)
Comment (DmytryjK) [reply submissions/DmytryjK/HTML-CSS-Popup/css/components/fonts.css]: fixed

File: submissions/DmytryjK/HTML-CSS-Popup/css/style.css (hunk 3 above)
Comment (DmytryjK) [reply submissions/DmytryjK/HTML-CSS-Popup/css/style.css]: fixed

File: submissions/DmytryjK/HTML-CSS-Popup/css/components/header.css (hunk 4 above)
Comment (A-Ostrovnyy) [reply submissions/DmytryjK/HTML-CSS-Popup/css/components/header.css:1]: Better upload all modules without compiled style.css.

**Comment by A-Ostrovnyy** [comment]
No need to add an empty file: [https://github.com/kottans/frontend-2022-homeworks/pull/37/files#diff-cb4e305b59c230dd9f2eb93110cd020f89c282589d220f4ea44f19689ab2f796](https://github.com/kottans/frontend-2022-homeworks/pull/37/files#diff-cb4e305b59c230dd9f2eb93110cd020f89c282589d220f4ea44f19689ab2f796)

**Comment by A-Ostrovnyy** [comment]
Check your work after uploading) 
image

**Comment by A-Ostrovnyy** [comment]
When you check all your files and you think your PR ready for review click re-request review
image

File: submissions/DmytryjK/HTML-CSS-Popup/css/components/header.css (hunk 4 above)
Comment (DmytryjK) [reply submissions/DmytryjK/HTML-CSS-Popup/css/components/header.css]: > When you check all your files and you think your PR ready for review click re-request review image

I hope i fixed all problems for my project. But not enough understend problem on this link  - https://github.com/kottans/frontend-2022-homeworks/pull/37/files#diff-cb4e305b59c230dd9f2eb93110cd020f89c282589d220f4ea44f19689ab2f796
What i have to do?

File: submissions/DmytryjK/HTML-CSS-Popup/sass/components/header.scss [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+            grid-template-columns: repeat(3, 1fr);
+            grid-template-rows: repeat(2, 1fr);
+            grid-auto-rows: 1fr;
+            align-items: center;
+            justify-items: center;
+            gap: 30px;
+            padding: 40px 0;
+        }   
+        &:focus + .header__popup-btn-text  {
+            background-color: lightblue;
```
Comment (A-Ostrovnyy) [inline submissions/DmytryjK/HTML-CSS-Popup/sass/components/header.scss:185]: If you use CSS custom properties for colors use this approach throughout the project.

File: submissions/DmytryjK/HTML-CSS-Popup/sass/style.scss [hunk 7]
```diff
@@ -0,0 +1,19 @@
+@import 'variables/variables.scss';
+@import 'components/fonts.scss';
+
+* {
```
Comment (A-Ostrovnyy) [inline submissions/DmytryjK/HTML-CSS-Popup/sass/style.scss:4]: Move these CSS rules into an independent module `base.scss` and leave here just imports. 
Hard to read code when CSS rules are mixed with Sass imports.

File: submissions/DmytryjK/HTML-CSS-Popup/index.html [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width,initial-scale=1">
+    <title>Hooli Popup</title>
+    <link rel="icon" type="image/x-icon" href="images/favicon.png">
+    <link rel="stylesheet" href="css/style.css">
+</head>
+<body>
+    <header class="header">
+        <div class="container">
+            <a class="header__logo-link" href="#">
```
Comment (A-Ostrovnyy) [inline submissions/DmytryjK/HTML-CSS-Popup/index.html:14]: The page logo should lead us to the home page.

File: submissions/DmytryjK/HTML-CSS-Popup/index.html [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+                    </li>
+                    <li class="header__items">
+                        <label class="header__popup-label" for="checkbox">
+                            <input class="header__popup-input" id="checkbox" type="checkbox">
+                            <span class="header__popup-btn-img"></span>
+                            <ul class="header__popup-submenu">
+                                <li class="header__popup-item">
+                                    <a class="header__popup-link" href="#">
+                                        <picture>
+                                            <source srcset="images/WEBP/calendar.webp" type="image/webp">
```
Comment (A-Ostrovnyy) [inline submissions/DmytryjK/HTML-CSS-Popup/index.html:33]: :fire:

File: submissions/DmytryjK/HTML-CSS-Popup/index.html [hunk 10]
```diff
@@ -0,0 +1,236 @@
...
+                                            <source srcset="images/calendar.png" type="image/jpeg"> 
+                                            <img class="header__popup-img" src="images/calendar.png" alt="calendar icon">
```
Comment (A-Ostrovnyy) [inline submissions/DmytryjK/HTML-CSS-Popup/index.html:35]: Read [this article](https://htmlacademy.ru/blog/articles/alt-text) about `alt` attribute.
Sorry for russian but it's a good one.

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Awwesome!
See my comments below.

File: submissions/DmytryjK/HTML-CSS-Popup/css/components/header.css (hunk 4 above)
Comment (A-Ostrovnyy) [reply submissions/DmytryjK/HTML-CSS-Popup/css/components/header.css]: You did everything right. Your PR should contain only source files, no need to include compiled files to your PR.

**Comment by DmytryjK** [comment]
I hope everything is ok :)
Is it good idea, if i committed every file that i add, or it's like a spam?

File: submissions/DmytryjK/HTML-CSS-Popup/index.html [hunk 11]
```diff
@@ -0,0 +1,236 @@
...
+            <nav class="header__navigation">
+                <ul class="header__list">
+                    <li class="header__items">
+                        <a class="header__links" href="#">Mail</a>
+                    </li>
+                    <li class="header__items">
+                        <a class="header__links" href="#">Images</a>
+                    </li>
+                    <li class="header__items">
+                        <label class="header__popup-label" for="checkbox">
```
Comment (A-Ostrovnyy) [inline submissions/DmytryjK/HTML-CSS-Popup/index.html:26]: When user opens a popup using keyboard and press the `Tab` key focus must go to the first interactive element inside the popup. In your case focus goes to the popup wrapper and after second click goes to the the Calendar icon.

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Almost done!
Final issue)

File: submissions/DmytryjK/HTML-CSS-Popup/index.html (hunk 11 above)
Comment (DmytryjK) [reply submissions/DmytryjK/HTML-CSS-Popup/index.html:26]: I try to use somesing like this:
```
input:checked ~ .header__popup-submenu > li:first-child > a:active  {
            outline: solid $color-state-first; 
        }
```
or 

```
input:focus ~ .header__popup-submenu > li:first-child > a:active  {
            outline: solid $color-state-first; 
        }
```
But it doesn't working :( :

Could you show me the way in which I should to move for solve this task?

Comment (A-Ostrovnyy) [reply submissions/DmytryjK/HTML-CSS-Popup/index.html:26]: When checkbox gets focus you highlight the label, what you put inside the label?)

Comment (DmytryjK) [reply submissions/DmytryjK/HTML-CSS-Popup/index.html:26]: My list items with links. Don't have any hypothesis. 
Maybe my structure isn't correctly?

```
<label>
     <input>
     ul>li>a...
</label>
```

Do i have to do something like this? 
```
<label></label>
<input>
ul>li>a...
```

This code change outline in the links icons, when i click on menu button, but won't change focus((

```
input:focus ~ .header__popup-submenu > li:first-child > a {
            outline: solid $color-state-first;
            border-radius: 5px;
} 
```

Comment (A-Ostrovnyy) [reply submissions/DmytryjK/HTML-CSS-Popup/index.html:26]: You need something like this:
```HTML
<input type="checkbox" >
<label></label>
<div class="popup></div>
```
In this case, when checkbox will be in focus, you can highlight the label and show popup)

Comment (DmytryjK) [reply submissions/DmytryjK/HTML-CSS-Popup/index.html:26]: Good night, sorry, but i really don't know, how to change focus from checked input to first link in navbar menu. I tried to find some information about this case, but didn't find anything :( Can we check this in discord?)

Comment (A-Ostrovnyy) [reply submissions/DmytryjK/HTML-CSS-Popup/index.html:26]: @DmytryjK I have already given you the correct answer to this question) Look carefully at the code example and compare it with your code)

Comment (DmytryjK) [reply submissions/DmytryjK/HTML-CSS-Popup/index.html:26]: I tried rewrite my code as in the example above. But it was the same story: focus remains on the menu button element. Honestly I don't understand what the different between:
```
<label>
 <input></input>
 <ul>
  <li></li>
 </ul>
</label>
```

and

```
<label></label>
<input></input>
<ul>
  <li></li>
</ul>
```

In both cases, when I press the menu button, submenu opens without problems. But the focus doesn't jump to the 1st element from the list.
I don't understand the logic of how exactly to change the target when clicking on the checkbox of the menu button.

Comment (DmytryjK) [reply submissions/DmytryjK/HTML-CSS-Popup/index.html]: Thanks for call. Issues have been resolved. Need clear cache in browser for correct review DEMO site.

**Review by A-Ostrovnyy — APPROVED** [review]
Well done!

**Pull request merged**